Directory.Build.targets

```diff
@@ -112,7 +110,7 @@
         '$(IsSampleProject)' != 'true' AND
         '$(IsSpecificationTestProject)' != 'true' AND
         '$(IsTestAssetProject)' != 'true' AND
-        '$(IsTestProject)' != 'true' AND
+        !$(IsTestProject) AND
```

Comment:
I'm personally not a fan of doing these literal boolean checks in _some_ places when we can't do it everywhere, especially in a block like this one where all of the other properties have to be checked literally. I can imagine this being confusing to somebody looking at this file for the first time

Reply:
The only place where we can't treat `$(IsTestProject)` or `$(IsUnitTestProject)` as `bool` is in Directory.Build.props above the Helix SDK import.

Is your objection about mixing `bool` checks w/ `string` comparisons in a single `Condition`❔ Someone hunting for `$(IsTestProject)` would also be confused if I left its treatment inconsistent (which it was before).

Reply:
Yes, it's about mixing `bool` checks with `string` comparisons in a single condition.

> Someone hunting for $(IsTestProject) would also be confused if I left its treatment inconsistent (which it was before).

I prefer that inconsistency to the one introduced by this PR, but not enough to block on it. I think I'd prefer not using `bool` comparisons for properties _anywhere_ until we can use it _everywhere_ (which functionally means never using those comparisons), but again, not enough to block on it.